### ISSUE TITLE
Table (td) doesn't respect max-width and text-overflow

### ISSUE THREAD

**Issue by kesara** [Kozea/WeasyPrint#2082]
WeasyPrint doesn't respect the `max-width` of `td` when there's a long word.
And WeasyPrint seems to ignore `overflow-wrap: break-word;` as well.

Example HTML
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>WeasyPrint table-width test</title>
    <style>
      body {
        max-width: 300px;
      }
      td {
        max-width: 200px;
      }
      .break-word {
        overflow-wrap: break-word;
      }
    </style>
  </head>
  <body>
    <h3>With very long word</h3>
    <p>Lorem ipsum dolor sit amet, verylongnamehere: Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</p>
    <h3>With very long word and overflow-wrap: break-word</h3>
    <p class="break-word">Lorem ipsum dolor sit amet, verylongnamehere: Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</p>
    <h3>Without a very long word</h3>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</p>
    <h3>Table with very long word</h3>
    <table>
      <tr>
        <td>Lorem ipsum dolor sit amet, verylongnamehere: Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</td>
      </tr>
    </table>
    <h3>Table with very long word and overflow-wrap: break-word</h3>
    <table>
      <tr>
        <td class="break-word">Lorem ipsum dolor sit amet, verylongnamehere: Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</td>
      </tr>
    </table>
    <h3>Table without a very long word</h3>
    <table>
      <tr>
        <td>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Imperdiet massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Nulla at volutpat diam ut venenatis.</td>
      </tr>
    </table>
  </body>
</html>
```

Expected output (from Firefox):
<img width="335" alt="Screenshot 2024-02-29 at 19 09 11" src="https://github.com/Kozea/WeasyPrint/assets/1027692/6a4ab946-314f-44b7-b9f7-5a899f899389">

WeasyPrint version 61.1 output:
<img width="385" alt="Screenshot 2024-02-29 at 19 09 33" src="https://github.com/Kozea/WeasyPrint/assets/1027692/ee2dd992-e18d-41cf-b2dc-284e08b741dd">

**Comment by liZe** [comment]
Closed by #2085.

**Comment by kesara** [comment]
Thanks for the quick fix @liZe.
